### PR TITLE
Adding push notification tests to helix

### PR DESCRIPTION
**Pull request by loneursid**
Experimenting with Helix

### CODE REVIEW

File: build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml [hunk 1]
```diff
@@ -134,6 +134,14 @@ jobs:
...
   #     arguments: -TestFilePattern 'mrtcore_test_binaries_x64\MrtCoreManagedTest\*.dll' -TestBinaryDirectoryPath '$(testPayloadDir)' -OutputProjFile '$(helixWorkItemsDir)\MrtCore-MrtCoreManagedTest.proj' -WorkItemPrefix 'MrtCore-MrtCoreManagedTest-$(buildPlatform)-$(buildConfiguration)' -TestNamePrefix 'MrtCore.MrtCoreManagedTest.$(buildPlatform).$(buildConfiguration)'
 
+  # Discover the TAEF Push Notifications test binaries to run and generate the Helix Work Items for them.
+  - task: powershell@2
+    displayName: 'Discover Push Notifications UnitTests'
+    inputs:
+      targetType: filePath
+      filePath: $(winUIHelixPipelineScripts)\GenerateHelixWorkItems.ps1
+      arguments: -TestFilePattern '$(Build.SourcesDirectory)\BuildOutput\$(buildConfiguration)\$(buildPlatform)\PushNotificationTests\*.dll' -TestBinaryDirectoryPath '$(testPayloadDir)' -OutputProjFile '$(helixWorkItemsDir)\PushNotificationTests.proj' -WorkItemPrefix 'MrtCore.MrmUnitTests.$(buildPlatform).$(buildConfiguration)' -TestNamePrefix 'PushNotificationTests.$(buildPlatform).$(buildConfiguration)'
+
```
Comment (DrusTheAxe) [inline build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml:144]: @jefgen have those aggregator pipeline test changes been replicated here yet? (I haven't done it)

And how about that data-driven approach we discussed? Waaay preferable to developers editing the pipeline for new tests, and with all the myriad fine print we discussed...

Comment (loneursid) [reply build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml:144]: @DrusTheAxe , @jefgen, That's an old side projects (adding the push notifications unit tests to the build pipeline) that I've just reactivated for EIM. I've got about two weeks left before going on vacation that I can work on this. If you care to brief me on the data -driven approach (or other plans) maybe I can be of some help.

File: build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml [hunk 2]
```diff
@@ -84,6 +84,49 @@ jobs:
...
+      dir $(Build.SourcesDirectory)\BuildOutput\Release\$(buildPlatform)
+      dir D:\
+      dir D:\a\
+      dir D:\a\1\
+      dir D:\a\1\s\
+      dir D:\a\1\s\BuildOutput\
+      dir D:\a\1\s\BuildOutput\Release\
+      dir D:\a\1\s\BuildOutput\Release\x64
+      dir D:\a\1\s\BuildOutput\Release\x64\PushNotificationTests
+    displayName: 'Copy push notification tests debug'
```
Comment (DrusTheAxe) [inline build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml:99]: Where's the 'copy'?

P.S. If this is debugging-printfery you may want to consider fewer/one `DIR /S`

File: build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml [hunk 3]
```diff
@@ -84,6 +84,49 @@ jobs:
...
+
+  # Copy and Publish the test binaries so we can run them in Helix later.
+  - task: CopyFiles@2
+    displayName: 'copy push notifications test binaries'
+    condition: and(succeeded(), or(eq(variables['buildPlatform'], 'x86'), eq(variables['buildPlatform'], 'x64')))
+    inputs:
+      SourceFolder: '$(Build.SourcesDirectory)\BuildOutput\Release\$(buildPlatform)'
+      Contents: |
+        PushNotificationTests\**
```
Comment (DrusTheAxe) [inline build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml:108]: I thought Helix infra was copying the entire BuildOutput/configuration/platform directory tree to the helix test environment. That's what the aggregator pipeline does for its testing

Comment (DrusTheAxe) [reply build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml:108]: @jefgen @alexlamtest

File: build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml (hunk 2 above)
Comment (loneursid) [reply build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml:91]: Yeah, that was debugging and I didn't think of using /s. 

So it looks like the entire solution is being built and I can see that the pushnotificationtests project is built and I would expect the output to be in $(Build.SourcesDirectory)\BuildOutput\Release\$(buildPlatform) but I didn't. The only thing in that directory is the helix test directory.

I'm sure that the buildoutput for pushtnotificationtests existed at some point, but by the time I get to executing the helix template, it's gone. The question is where /when does it get wiped out? I couold grab a copy before then and then use that copy to publish it for helix.

Comment (loneursid) [reply build/AzurePipelinesTemplates/WindowsAppSDK-RunHelixTests-Job.yml:91]: The copy is just after that chunck of debuging printfery code. I was expecting to be able to copy the pushnotifitcationtests folder under BuildOutput\Release\x64 or x86 depending but the pushnotifitcationtests folder doesn't seem to exist anymore at this point in the build process.

**Comment by loneursid** [comment]
Closing this experiment. Will start a new one using the latest changes made to the build pipelines

**Pull request closed without merging**